### DISCUSSION TITLE
docusaurus Minified React error #300

### DISCUSSION THREAD

**Discussion by Chasen-Zhang** [facebook/docusaurus#8700] [Q&A]
this is my code, I'm just filtering the list by keyword, Local development is ok, but online is not
![image](https://user-images.githubusercontent.com/15354455/220833643-7008901b-8047-4e69-b4e1-57d4ad50e37a.png)

![image](https://user-images.githubusercontent.com/15354455/220833386-314d5308-670b-4b32-a9d6-3fad4bc573c6.png)

https://reactjs.org/docs/error-decoder.html?invariant=300

**Comment by slorber** [comment]
The error message from the url above:

> Rendered fewer hooks than expected. This may be caused by an accidental early return statement.

Run this in local dev to get the error locally. Your code likely doesn't respect the React rules of hooks.

It's out of the scope of Docusaurus, and impossible to help without seeing your whole code.

You can install the ESLint React rules of hooks plugin so that it tells you bad hooks usage

**Reply by Chasen-Zhang** [reply]
I don't think it's hooks. I wrote it that way at the beginning, It's the same mistake
![image](https://user-images.githubusercontent.com/15354455/221488804-1432c152-ba43-4fea-9d4b-49865de71783.png)

 @slorber

**Comment by Chasen-Zhang** [comment]


**Reply by slorber** [reply]
You are rendering `useBaseUrl` inside a JSX condition so this is definitively not respecting the rules of React hooks.

![CleanShot 2023-03-01 at 17 11 53@2x](https://user-images.githubusercontent.com/749374/222197155-d17520b3-cddd-4afc-a1da-7cef9963ca3a.png)

**Reply by Chasen-Zhang** [reply]
Thanks! LOVE YOU AGAIN! @slorber
